src/safeposix/syscalls/sys_calls.rs

```diff
@@ -258,6 +271,20 @@ impl Cage {
         //may not be removable in case of lindrustfinalize, we don't unwrap the remove result
         interface::cagetable_remove(self.cageid);
 
+        // if the cage has parent
+        if self.parent != self.cageid {
+            let parent_cage = interface::cagetable_getref_opt(self.parent);
+            // if parent hasn't exited yet
+            if let Some(parent) = parent_cage {
```

Comment:
Not some we implemented in lind-nacl either but worth adding a BUG tag here since we dont handle the situation where a parent has exited already. I'm not sure if we want to implement zombies being reclaimed by another process.